subsys/net/lib/lwm2m/lwm2m_obj_connmon.c

```diff
@@ -128,6 +143,14 @@ static struct lwm2m_engine_obj_inst *connmon_create(uint16_t obj_inst_id)
 	link_quality = 0U;
 	mnc = 0U;
 	mcc = 0U;
+#if defined(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_2) ||                                            \
+	defined(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_3)
+	snr = 0U;
+	lac = 0U;
+#endif /* CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_2 || CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_3 */
+#if defined(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_3)
+	cel = 0U;
+#endif /* CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_3 */
 
```

Comment:
Another thing to consider:
If we write the code using this way instead:
```c
/* Declare static variable unconditionally */
static int32_t snr;

...
/* in connmon_create(uint16_t obj_inst_id) */
if (IS_ENABLED(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_2)) {
        snr = 0U;
}

...

if (IS_ENABLED(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_2) ||
    IS_ENABLED(CONFIG_LWM2M_CONNMON_OBJECT_VERSION_1_2)) {
        INIT_OBJ_REST_DATA(...);
}
```

I believe it drops the unused static variables away. Code becomes cleaner.

Reply:
@SeppoTakalo : You are right and I have a different proposal to remove the crazy ifdefness :)